docs/source/optim.rst

```diff
@@ -129,6 +129,50 @@ Algorithms
     Rprop
     SGD
 
+Many of our algorithms have various implementations optimized for performance, so
```

Comment:
nit: they're not all optimized for performance. Some are performance, some readability/generality?